packages/modules/common/configurable_tariff.py

```diff
@@ -26,21 +32,91 @@ def __init__(self,
-                    self.fault_state.no_error(
-                        f'Die Preisliste hat nicht 24, sondern {len(tariff_state.prices)} Einträge. '
-                        'Die Strompreise werden vom Anbieter erst um 14:00 für den Folgetag aktualisiert.')
-
-    def _remove_outdated_prices(self, tariff_state: TariffState) -> TariffState:
-        current_hour = str(int(create_unix_timestamp_current_full_hour()))
-        for timestamp in list(tariff_state.prices.keys()):
-            if timestamp < current_hour:
+                self.__query_et_provider_data_once_per_day()
+                timeslot_length_seconds = self.__calculate_price_timeslot_length()
+                self.__tariff_state = self._remove_outdated_prices(self.__tariff_state, timeslot_length_seconds)
+                self.__store_and_publish_updated_data()
+                self.__log_and_publish_progress(timeslot_length_seconds)
+
+    def __query_et_provider_data_once_per_day(self):
+        if datetime.now() > self.__next_query_time:
+            log.info(f'Wartezeit {self.__next_query_time.strftime("%Y%m%d-%H:%M:%S")}'
+                     ' abgelaufen, Strompreise werden abgefragt')
+            try:
+                last_known_timestamp = self.__get_last_entry_time_stamp()
+                self.__tariff_state = self._component_updater()
+                if last_known_timestamp < (max(self.__tariff_state.prices)):
+                    self.__calulate_next_query_time()
+                else:
+                    log.info('Keine Daten für morgen erhalten, weiterer Versuch in 5 Minuten')
+            except Exception as e:
                 self.fault_state.warning(
-                    'Die Preisliste startet nicht mit der aktuellen Stunde. Abgelaufene Einträge wurden entfernt.')
-                tariff_state.prices.pop(timestamp)
+                    f'Fehler beim Abruf der Strompreise: {e}, nächster Versuch in 5 Minuten.'
+                )
+            log.info(f'Nächster Abruf der Strompreise frühestens {self.__next_query_time.strftime("%Y%m%d-%H:%M:%S")}.')
+
+    def __day_of(self, date: datetime) -> datetime:
+        return date.replace(hour=0, minute=0, second=0, microsecond=0)
+
+    def __next_query_message(self) -> str:
+        tomorrow = (
+            ''
+            if self.__day_of(datetime.now()) == self.__day_of(self.__next_query_time)
+            else 'morgen '
+        )
+        return (
+                f'frühestens {tomorrow}{self.__next_query_time.strftime("%H:%M")}'
+                if datetime.now() < self.__next_query_time
+                else "im nächsten Regelzyklus"
+        )
+
+    def __log_and_publish_progress(self, timeslot_length_seconds):
+        def publish_info(message_extension: str) -> None:
+            self.fault_state.no_error(
+                        f'Die Preisliste hat {message_extension}{len(self.__tariff_state.prices)} Einträge. '
+                        f'Nächster Abruf der Strompreise {self.__next_query_message()}.')
```

Comment:
> @LKuemmel Spricht etwas dagegen, bei "fixed-hours" die Preise für zwei Tage zu ermitteln? Oder die Preise nicht ab letzter Mitternacht sondern der aktuellen Stunde zu erzeugen? Ich meine, dass abgelaufene Preise eh verworfen werden?

Das hatten wir bisher implizit, weil die Tarife jede Stunde angefordert wurden. So hatte "fixed hours" immer 24 Einträge. Das passiert jetzt aber nicht mehr. 
Darüber hinaus  gibt es den Hinweis auf die Aktualisierungszeit mit diesem PR sowieso, auch wenn mehr Daten zur Verfügung stehen. 

Die Frage ist, ob ich den noch weg machen (also zwischen "echten" Tarifen und berechneten unterscheiden) soll. Vielleicht reicht aber auch 'ne allgemeinere Formulierung:

```suggestion
                        f'Nächste Aktualisierung der Strompreise {self.__next_query_message()}.')
```